### PR TITLE
PADV-978 -  Create Classes table in Classes View

### PR DESCRIPTION
**Pull request by sergivalero20**
### Ticket

https://agile-jira.pearson.com/browse/PADV-978

### Description
Add classes table and classes view.

### Changes made

- [X] Add classes option is in the sidebar.
- [X] Create Table with respective fields.
- [X] Add pagination to the Table.
- [x] Create tests.

### Screenshots
Classes page 1:

image

Classes page 2 (see **instructor column**):
image

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1980/classes

### CODE REVIEW

File: src/features/Classes/ClassesPage/index.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import Container from '@edx/paragon/dist/Container';
+import { Pagination } from '@edx/paragon';
```
Comment (AuraAlba) [inline src/features/Classes/ClassesPage/index.jsx:5]:
```suggestion
import { Pagination, Container } from '@edx/paragon';
```
Please move Container import with Pagination

File: src/features/Classes/ClassesTable/columns.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    accessor: 'numberOfStudents',
+  },
+  {
+    Header: 'Max',
+    accessor: 'maxStudents',
+  },
+  {
+    Header: 'Instructors',
+    accessor: ({ instructors }) => (
+      <ul style={{ listStyleType: 'none', paddingLeft: 0 }}>
```
Comment (AuraAlba) [inline src/features/Classes/ClassesTable/columns.jsx:35]: Please create a stylesheet for this component, define a className for this element and move the styles there

File: src/features/Classes/ClassesTable/index.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import { IntlProvider } from 'react-intl';
+import { Row, Col } from '@edx/paragon';
+import DataTable from '@edx/paragon/dist/DataTable';
```
Comment (AuraAlba) [inline src/features/Classes/ClassesTable/index.jsx:6]:
```suggestion
import { Row, Col, Datatable } from '@edx/paragon';
```
Move Datable import with others

File: src/features/Classes/data/slice.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      const { results, count, numPages } = payload;
+      state.table.status = RequestStatus.SUCCESS;
+      state.table.data = results;
+      state.table.numPages = numPages;
+      state.table.count = count;
+    },
+    fetchClassesDataFailed: (state) => {
+      state.table.status = RequestStatus.ERROR;
+    },
+    updateFilters: (state, { payload }) => {
```
Comment (AuraAlba) [inline src/features/Classes/data/slice.js:37]: Remove this reducer because is not part of the scope

**Review by AuraAlba — CHANGES_REQUESTED** [review]

File: src/features/Classes/ClassesTable/index.jsx (hunk 3 above)
Comment (sergivalero20) [reply src/features/Classes/ClassesTable/index.jsx:6]: @AuraAlba I got and error if I try to do this.

File: src/features/Classes/ClassesTable/_test_/columns.test.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    expect(endDateColumn).toHaveProperty('accessor', 'endDate');
+
+    expect(numberOfStudentsColumn).toHaveProperty('Header', 'Students Enrolled');
+    expect(numberOfStudentsColumn).toHaveProperty('accessor', 'numberOfStudents');
+
+    expect(maxStudentsColumn).toHaveProperty('Header', 'Max');
+    expect(maxStudentsColumn).toHaveProperty('accessor', 'maxStudents');
+
+    expect(instructorsColumn).toHaveProperty('Header', 'Instructors');
+    expect(instructorsColumn).toHaveProperty('accessor', 'instructors');
```
Comment (anfbermudezme) [inline src/features/Classes/ClassesTable/_test_/columns.test.jsx:37]: Tests are failing for this line. @AuraAlba

**Review by anfbermudezme — CHANGES_REQUESTED** [review]

**Review by AuraAlba — APPROVED** [review]

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**